### PR TITLE
Add support for `default-groups`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This PR adds support for `tool.uv.default-groups`, which defaults to `["dev"]` for backwards-compatibility. These represent the groups we sync by default.

### CODE REVIEW

File: crates/uv-configuration/src/dev.rs [hunk 1]
```diff
@@ -121,70 +131,120 @@ impl DevGroupsSpecification {
...
+
+/// The manifest of `dependency-groups` to include, taking into account the user-provided
+/// [`DevGroupsSpecification`] and the project-specific default groups.
+#[derive(Debug, Clone)]
+pub struct DevGroupsManifest {
+    /// The specification for the development dependencies.
+    pub(crate) spec: DevGroupsSpecification,
+    /// The default groups to include.
+    pub(crate) defaults: Vec<GroupName>,
+}
```
Comment (charliermarsh) [inline crates/uv-configuration/src/dev.rs:210]: I don't feel strongly on the name here, but this is the specification along with the project-specific defaults.

**Comment by zanieb** [comment]
Yeah I was imagining adding `--no-default-groups`, `--no-group` seems fine too.

File: crates/uv/src/commands/project/run.rs [hunk 2]
```diff
@@ -563,6 +549,44 @@ pub(crate) async fn run(
...
+                            .is_some_and(|groups| groups.contains_key(group))
+                        {
+                            return Err(anyhow::anyhow!(
+                                "Default group `{group}` is not defined in the project's `dependency-group` table"
+                            ));
+                        }
+                    }
+                    defaults.clone()
+                } else {
+                    vec![DEV_DEPENDENCIES.clone()]
```
Comment (zanieb) [inline crates/uv/src/commands/project/run.rs:587]: Can we avoid repeating this block in each command?

File: crates/uv/tests/it/sync.rs [hunk 3]
```diff
@@ -1273,6 +1273,143 @@ fn sync_non_existent_group() -> Result<()> {
...
+
+    context.lock().assert().success();
+
+    uv_snapshot!(context.filters(), context.sync(), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: Default group `bar` is not defined in the project's `dependency-group` table
```
Comment (zanieb) [inline crates/uv/tests/it/sync.rs:1305]: Perhaps something like

```suggestion
    error: Default group `bar` (from `tool.uv.default-groups`) is not defined in the project's `dependency-group` table
```

File: crates/uv/src/commands/project/run.rs (hunk 2 above)
Comment (charliermarsh) [reply crates/uv/src/commands/project/run.rs]: Done

File: crates/uv/tests/it/sync.rs (hunk 3 above)
Comment (charliermarsh) [reply crates/uv/tests/it/sync.rs:1305]: Done

**Review by zanieb — APPROVED** [review]

**Pull request merged**